weather.php

```diff
@@ -1,30 +1,48 @@
 <?php
-  /**
-   * This page talks to weatherapi.com.
-   */
 
-  // API URL.
-  // You can register for free to get your own api key (fb0... below). 
-  $url = "http://api.weatherapi.com/v1/current.json?key=fb0b4c5c22d04c22be2202032210112&aqi=no&q=";
+/**
+ * This page talks to weatherapi.com.
+ */
 
-  // Check and filter query parameters.
-  $query = filter_input(INPUT_GET, 'query', FILTER_DEFAULT);
+// API URL.
+// You can register for free to get your own api key (fb0... below). 
+// forecastUrl returns both current and forecast weather conditions
+// $currentUrl = "http://api.weatherapi.com/v1/current.json?key=fb0b4c5c22d04c22be2202032210112&aqi=no&q=";
+$forecastUrl = "http://api.weatherapi.com/v1/forecast.json?key=fb0b4c5c22d04c22be2202032210112&aqi=no&days=3&q=";
 
-  if ($query) {
-    $query = urlencode($query);
+// Parse query parameters.
+$input = filter_input(INPUT_GET, 'query');
+// Verify input is INT < 99999
+$query = filter_var($input, FILTER_VALIDATE_INT, ["options" => ["max_range" => 99999]]);
```

Comment:
It's not accepting city, state because it's expecting only an int (zip code).

Reply:
Corrected this in most recent push. Custom regex rather than pre-built php filters.